### PR TITLE
fix: deduplicate entries in children and references

### PR DESCRIPTION
**Pull request by dandhlee**
Before you open a pull request, note that this repository is forked from [here](https://github.com/docascode/sphinx-docfx-yaml/).
Unless the issue you're trying to solve is unique to this specific repository, 
please file an issue and/or send changes upstream to the original as well.

__________________________________________________________________

There was an issue with the way Sphinx imports docstrings, and processes certain items twice in `process_docstring` function. Instead of checking for duplicate entries in the children/reference, we'll check to see if we've already processed a docstring or not and solve it from the root cause.

Fixes internally filed issue

> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: docfx_yaml/extension.py [hunk 1]
```diff
@@ -626,6 +628,15 @@ def process_docstring(app, _type, name, obj, options, lines):
     """
     This function takes the docstring and indexes it into memory.
     """
+
+    # Check if we already processed this docstring.
+    if name not in app.env.docfx_uid_names:
+        # Checking in dictionary("{}") takes O(1) average time for a bit more space,
+        # whereas checking in lists("[]") takes O(n) average time.
```
Comment (tbpg) [inline docfx_yaml/extension.py:635]: Not sure we need this, but it doesn't hurt. Using a dict as a set is standard.

File: docfx_yaml/extension.py [hunk 2]
```diff
@@ -626,6 +628,15 @@ def process_docstring(app, _type, name, obj, options, lines):
     """
     This function takes the docstring and indexes it into memory.
     """
+
+    # Check if we already processed this docstring.
+    if name not in app.env.docfx_uid_names:
```
Comment (tbpg) [inline docfx_yaml/extension.py:633]: I would flip this, returning early if it's already processed. Then, you don't need the `else`.

File: docfx_yaml/extension.py (hunk 1 above)
Comment (dandhlee) [reply docfx_yaml/extension.py:635]: done.

File: docfx_yaml/extension.py (hunk 2 above)
Comment (dandhlee) [reply docfx_yaml/extension.py]: done!

**Comment by dandhlee** [comment]
Thank you! Ready for review :)

**Review by tbpg — APPROVED** [review]

**Pull request merged**